shell/platform/embedder/embedder.h

```diff
@@ -1740,17 +1760,30 @@ typedef struct {
   /// immediately after the root isolate has been created and marked runnable.
   VoidCallback root_isolate_create_callback;
   /// The callback invoked by the engine in order to give the embedder the
-  /// chance to respond to semantics node updates from the Dart application.
+  /// chance to respond to updates to semantics nodes and custom actions from
+  /// the Dart application.
+  ///
+  /// The callback will be invoked on the thread on which the `FlutterEngineRun`
+  /// call is made.
+  ///
+  /// If this callback is provided, update_semantics_node_callback and
+  /// update_semantics_custom_action_callback must not be provided.
+  FlutterUpdateSemanticsCallback update_semantics_callback;
```

Comment:
This needs to go at the end of the struct or it will be an ABI-breaking change.

Reply:
Very nice catch, fixed!